domenetjenester/vedtak/src/main/java/no/nav/ung/sak/hendelse/vedtak/VurderManglendeKontrollAvPeriode.java

```diff
@@ -0,0 +1,54 @@
+package no.nav.ung.sak.hendelse.vedtak;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.Dependent;
+import jakarta.inject.Inject;
+import no.nav.abakus.vedtak.ytelse.Ytelse;
+import no.nav.abakus.vedtak.ytelse.v1.YtelseV1;
+import no.nav.ung.kodeverk.behandling.BehandlingÅrsakType;
+import no.nav.ung.kodeverk.behandling.FagsakYtelseType;
+import no.nav.ung.sak.behandlingskontroll.FagsakYtelseTypeRef;
+import no.nav.ung.sak.behandlingslager.behandling.Behandling;
+import no.nav.ung.sak.behandlingslager.behandling.repository.BehandlingRepository;
+import no.nav.ung.sak.behandlingslager.fagsak.FagsakRepository;
+import no.nav.ung.sak.typer.Saksnummer;
+import no.nav.ung.sak.ytelse.kontroll.ManglendeKontrollperioderTjeneste;
+
+import java.util.List;
+
+@ApplicationScoped
+@FagsakYtelseTypeRef(FagsakYtelseType.UNGDOMSYTELSE)
+public class VurderManglendeKontrollAvPeriode implements VurderOmVedtakPåvirkerSakerTjeneste {
+
+    private ManglendeKontrollperioderTjeneste manglendeKontrollperioderTjeneste;
+    private FagsakRepository fagsakRepository;
+    private BehandlingRepository behandlingRepository;
+
+    public VurderManglendeKontrollAvPeriode() {
+    }
+
+    @Inject
+    public VurderManglendeKontrollAvPeriode(ManglendeKontrollperioderTjeneste manglendeKontrollperioderTjeneste, FagsakRepository fagsakRepository, BehandlingRepository behandlingRepository) {
+        this.manglendeKontrollperioderTjeneste = manglendeKontrollperioderTjeneste;
+        this.fagsakRepository = fagsakRepository;
+        this.behandlingRepository = behandlingRepository;
+    }
+
+    @Override
+    public List<SakMedPeriode> utledSakerMedPerioderSomErKanVærePåvirket(Ytelse vedtakHendelse) {
+        var saksnummer = new Saksnummer(vedtakHendelse.getSaksnummer());
+        var fagsak = fagsakRepository.hentSakGittSaksnummer(saksnummer).orElseThrow();
+        Behandling vedtattBehandling = behandlingRepository.hentBehandling(((YtelseV1) vedtakHendelse).getVedtakReferanse());
```

Comment:
trenger du å hente behandling her? Ser ut som vedtakHendelse.vedtakReferanse er behandlingId (siden det er det som brukes i hentBehandling) som kan brukes rett i finnPerioderForManglendeKontroll.

Reply:
Vedtakreferanse er behandlingUUID, eg tenkjer at vi burde bruke behandlingId for intern logikk, er du enig?

Reply:
forsåvidt enig. Men å hente opp behandling bare for å mappe til behandlingId virker tungvint. Kan evt. sende med behandling eller behandlingId fra kaller...

Reply:
Men det vil brekke interfacet... Vet ikke hvor mye det interfacet er verdt hvis det bare er 1 impl

Reply:
Det er kanskje meir for at det skal vere likt som på k9-sak for å enklere kunne kjenne igjen konsepter

Reply:
Eg synes ikkje at det er så krise å hente ut behandling, men om vi skal beholde interface med ein implementasjon er jo eit godt poeng.